### ISSUE TITLE
Success on some GPU nodes, fail on other GPU nodesCheck failed: e == cudaSuccess CUDA: exclusive-thread device already in use by a different thread

### ISSUE THREAD

**Issue by ylqfp** [apache/incubator-mxnet#2632]
Hi all,
      When running standalone train_mnist.py example on GPU cluster, we always got a strange problem. Assuming nodes' names are GPU01 ~ GPU10, total 10 nodes. Sometimes, only GPU10 successed, sometimes GPU09. Others always fail.
      We located at the crashing code, finding that, they fail in step model.fit in "train_model.py":

``` python
model.fit(
X = train,
eval_data = val,
eval_metric = eval_metrics,
...
)
```

The error information is:

>  2016-07-06 09:59:34,752 Node[0] start with arguments Namespace(batch_size=128, data_dir='mnist/', gpus='0', kv_store='local', load_epoch=None, lr=0.1, lr_factor=1, lr_factor_epoch=1, model_prefix=None, network='lenet', num_epochs=10, num_examples=60000, save_model_prefix=None)
> [09:59:36] /home/wenjing/software/mxnet/src/io/iter_mnist.cc:94: MNISTIter: load 60000 images, shuffle=1, shape=(128,1,28,28)
> [09:59:36] /home/wenjing/software/mxnet/src/io/iter_mnist.cc:94: MNISTIter: load 10000 images, shuffle=1, shape=(128,1,28,28)
> 2016-07-06 09:59:36,240 Node[0] Start training with [gpu(0)]
> [09:59:36] [09:59:36] /home/wenjing/software/mxnet/dmlc-core/include/dmlc/./logging.h:235: [09:59:36] /home/wenjing/software/mxnet/mshadow/mshadow/./tensor_gpu-inl.h:35: Check failed: e == cudaSuccess CUDA: exclusive-thread device already in use by a different thread
> /home/wenjing/software/mxnet/dmlc-core/include/dmlc/./logging.h:235: [09:59:36] /home/wenjing/software/mxnet/mshadow/mshadow/./tensor_gpu-inl.h:35: Check failed: e == cudaSuccess CUDA: exclusive-thread device already in use by a different thread
> terminate called recursively
> terminate called after throwing an instance of 'dmlc::Error'

It seems that the device has already been used. However, we checked and checked again, and ensure that there was no other process using GPU .
Did anyone meet with the same problem?
Many Thanks!

**Comment by ylqfp** [comment]
@baiyancheng20

**Comment by ylqfp** [comment]
@mawenjing

**Comment by piiswrong** [comment]
@mli

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
